File: pgm/ABC_Visual_Basic/ABC_146_B.vb
' Imports System.Collections.Generic
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Sub Main()
        Dim N As Integer = Console.ReadLine()
        Dim S As String = Console.ReadLine()
        Dim alp As String = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        Console.WriteLine(Trans(S, alp, alp.Substring(N) + alp.Substring(0, N)))
        Return
    End Sub

    Shared Function Trans(ByVal S As String, ByVal f As String, ByVal t As String) As String
        For i As Integer = 0 To S.Length - 1
            Dim idx = f.indexOf(S(i))
            S = S.Substring(0, i) + t(idx) + S.Substring(i + 1)
        Next
        Trans = S
    End Function
End Class

File: pgm/ABC_C#/ABC_146_B.cs
using System;
// using System.Collections.Generic;
// using System.Linq;
using System.Text;

class Program
{
    static void Main()
    {
        int N = int.Parse(Console.ReadLine());
        string S = Console.ReadLine();
        string alp = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        Console.WriteLine(tran(S, alp, alp.Substring(N) + alp.Substring(0, N)));
        return;
    }
    static string tran(string S, string f, string t)
    {
        StringBuilder sb = new StringBuilder(S.Length);
        for (int i = 0; i < S.Length; i++)
        {
            int idx = f.IndexOf(S[i]);
            sb.Append(t[idx]);
        }
        return sb.ToString();
    }
}